creator/data_templates/field_definitions_schema.py

```diff
@@ -0,0 +1,243 @@
+        missing=False
+    )
+    accepted_values = fields.List(
+        fields.Raw(),
+        description="The list of acceptable values this field may be"
+                    "populated with",
+        # This says, if accepted_values is not None, then make sure it has
+        # at least one member in the list
+        validate=validate.Length(min=1),
+        missing=None,
+        allow_none=True
+    )
+    missing_values = fields.List(
+        fields.String(allow_none=True),
+        description="Values that a submitter may use if the data for this "
+        "field is unknown or unavailable for whatever reason",
+        # This says, if missing_values is not None, then make sure it has
+        # at least one member in the list
+        validate=validate.Length(min=1),
+
+        # If this is None, then we will interpret this to mean there are no
+        # defined missing values
+        missing=None,
+        allow_none=True
+    )
+    description = fields.String(
+        description="Description for the field",
+        required=True
+    )
+    instructions = fields.String(
+        description="Instructions or best practices for the data submitter "
+                    "on how to populate this field",
+        missing=None,
+        allow_none=True
+    )
+
+    class Meta:
+        # Maintain ordering of keys
+        ordered = True
+        # Exclude unknown fields from output
+        unknown = EXCLUDE
+
+    @pre_load
+    def clean(self, in_data, **kwargs):
+        """
+        Do our best to clean and correct any human errors or variability in
+        the input data before the it is validated against the schema
+        """
+        out_data = {}
+        for k, v in in_data.items():
+            # Title case to snake case
+            k = "_".join(k.strip().lower().split(" "))
+
+            # Strip white space
+            if isinstance(v, str):
+                v = v.strip() or None  # interpret empty str as null
+
+            # Convert str synonyms to booleans
+            if k == "required":
+                out_data[k] = coerce_bool(v)
+
+            # Lower case value in data_type
+            elif k == "data_type":
+                if not v:
+                    v = self.fields[k].missing
+                out_data[k] = str(v).strip().lower()
+
+            # Convert delimited str to list
+            elif k == "accepted_values" and isinstance(v, str):
+                values = multisplit(v, [",", ";", "\n"])
+                out_data[k] = [val.strip() for val in values if val.strip()]
+
+            elif k == "missing_values" and isinstance(v, str):
+                values = multisplit(v, [",", ";", "\n"])
+                out_data[k] = [val.strip() for val in values if val.strip()]
+            else:
+                out_data[k] = v
+
+        if out_data.get("accepted_values"):
+            out_data["data_type"] = "enum"
+
+        return out_data
+
+
+class FieldDefinitionsSchema(Schema):
```

Comment:
I would suggest adding a version number for the schema. If you do that, you can make changes to the schema in the future and then bump the version number. Thus you will have an easy way to fork logic or to identify schemas that need to be updated.

Reply:
I'm a little confused. I think you're suggesting adding a version to track changes to a template's field definitions? If so, do you mean add some sort of semantic version number field to the TemplateVersion model? 

If it is the latter, I'm a little hesitant to add that now bc we have no idea about who or what and how that version would be updated. Should the template admin get to update it however they see fit? Or should study creator have some rules for how to update the version (e.g. add/remove col -> major version update, desc change -> minor update)

Reply:
The version number would be authored by us whenever the schema here changes. Perhaps you add a new field to the `FieldDefinitionSchema`, then you would bump the version number. That way consumers of the data would know if they need to process the data differently.

Reply:
Ok I see. Does that mean we need a way to expose `FieldDefinitionSchema` in some way to users on the frontend? I think there is a third party library that can turn marshmallow schemas into json-schema